src/pages/fiscal-years/forms/PeriodsForm.js

```diff
@@ -0,0 +1,185 @@
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { DataSets } from 'src/resources/DataSets'
+import { formatDateFromApi } from 'src/lib/date-helper'
+import { ControlContext } from 'src/providers/ControlContext'
+
+const PeriodsForm = ({ labels, maxAccess, store }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const { platformLabels } = useContext(ControlContext)
+
+  const { recordId } = store
+
+  const editMode = !!recordId
+
+  const post = obj => {
+    const data = {
+      fiscalYear: recordId,
+      periods: obj.map(({ id, periodId, ...rest }) => ({
+        periodId: id,
+        ...rest
+      }))
+    }
+    postRequest({
+      extension: SystemRepository.FiscalPeriodPack.set2,
+      record: JSON.stringify(data)
+    })
+      .then(res => {
+        if (res) toast.success(platformLabels.Edited)
+        getPeriods()
+      })
+      .catch(error => {})
+  }
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId,
+      periods: []
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      periods: yup
+        .array()
+        .of(
+          yup.object().shape({
+            statusName: yup.string().required(' '),
+            startDate: yup.string().required(' '),
+            endDate: yup.string().required(' ')
+          })
+        )
+        .required(' ')
+    }),
+    onSubmit: values => {
+      post(values.periods)
+    }
+  })
+
+  const columns = [
+    {
+      component: 'numberfield',
+      label: labels?.period,
+      name: 'periodId',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'date',
+      name: 'startDate',
+      label: labels?.startDate
+    },
+    {
+      component: 'date',
+      name: 'endDate',
+      label: labels?.endDate
+    },
+    {
+      component: 'resourcecombobox',
+      label: labels?.status,
+      name: 'statusName',
+      props: {
+        datasetId: DataSets.FY_PERIOD_STATUS,
+        valueField: 'key',
+        displayField: 'value',
+        widthDropDown: 200,
+        mapping: [
+          { from: 'key', to: 'status' },
+          { from: 'value', to: 'statusName' }
+        ],
+        columnsInDropDown: [{ key: 'value', value: 'Name' }],
+        displayFieldWidth: 2
+      }
+    },
+    {
+      component: 'button',
+      name: 'saved',
+      onClick: (e, row) => {
+        stack({
+          Component: PeriodsModuleForm,
+          props: {
+            labels: labels,
+            maxAccess: maxAccess,
+            row,
+            recordId,
+            editMode: editMode
+          },
+          width: 600,
+
+          title: labels?.period
+        })
+      }
+    }
+  ]
+
+  useEffect(() => {
+    recordId && getPeriods()
+  }, [recordId])
+
+  const getPeriods = () => {
+    const defaultParams = `_fiscalYear=${recordId}`
```

Comment:
`const defaultParams = `_fiscalYear=${recordId}`
 var parameters = defaultParams`
 remove and assign `_fiscalYear=${recordId} `directly to parameters